### ISSUE TITLE
Wrap simplejson.errors.JSONDecodeError inside SmartcarException

### ISSUE THREAD

**Issue by dbartenstein** [smartcar/python-sdk#79]
On September 10th, 2020 between 7 p.m. and 8 p.m. CEST we faced issues when using Smartcar’s API to communicate with Tesla vehicles. After five minutes a `simplejson.errors.JSONDecodeError` exception was thrown.

Wouldn’t it be nicer to have that exception wrapped in a `SmartcarException` with additional information about what caused the error?

```
  File "/opt/.virtualenvs/django22/lib/python3.7/site-packages/smartcar/vehicle.py", line 311, in batch
    response = self.api.batch(requests)
  File "/opt/.virtualenvs/django22/lib/python3.7/site-packages/smartcar/api.py", line 79, in batch
    return requester.call('POST', url, json=json, headers=headers)
  File "/opt/.virtualenvs/django22/lib/python3.7/site-packages/smartcar/requester.py", line 31, in call
    body = response.json()
  File "/opt/.virtualenvs/django22/lib/python3.7/site-packages/requests/models.py", line 898, in json
    return complexjson.loads(self.text, **kwargs)
  File "/opt/.virtualenvs/django22/lib/python3.7/site-packages/simplejson/__init__.py", line 525, in loads
    return _default_decoder.decode(s)
  File "/opt/.virtualenvs/django22/lib/python3.7/site-packages/simplejson/decoder.py", line 370, in decode
    obj, end = self.raw_decode(s)
  File "/opt/.virtualenvs/django22/lib/python3.7/site-packages/simplejson/decoder.py", line 400, in raw_decode
    return self.scan_once(s, idx=_w(s, idx).end())
simplejson.errors.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

**Comment by dbartenstein** [comment]
### Addition

Very likely the request did not return a JSON body after five minutes, thus `body = response.json()` failed and the `simplejson.errors.JSONDecodeError` was raised.
https://github.com/smartcar/python-sdk/blob/master/smartcar/requester.py#L31

- What were the responses’ HTTP status codes of the requests that failed on September 10th 7 p.m. - 8 p.m.? It might be possible to skip `response.json()` for certain status codes or wrap it inside a try - except block?

### Background

As a developer I want to call actions via Smartcar’s Python SDK and not having to catch any other exceptions that those derived from `SmartcarException`.

**Comment by sankethkatta** [comment]
Hello @dbartenstein, thanks for raising this issue. During that brief window you mentioned, Smartcar's API had an issue where a small subset of errors were not returning the our standard JSON error format. The issue has since been fixed and we do not expect this situation to arise again.

That being said, we like your suggestion to have all SDK errors bubble up to the developer as a `SmartcarException`. We will put up a PR to implement this and we'll leave this issue open until it is merged.

**Comment by sankethkatta** [comment]
Fixed by #84
